From 65aea672e0ff3f4c11e71a4f33676fadab48223f Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 21 Jul 2020 09:17:10 -0700
Subject: [PATCH 1452/1852] v4l: xilinx: dprx: Fix uninitialized variables

commit d6a92ea993d7cf1a26dcbbf8e78a4b2431689909 from
https://github.com/Xilinx/linux-xlnx.git

It's single pad, so the pad idx can be removed.

drivers/media/platform/xilinx/xilinx-dprxss.c: In function 'xdprxss_init_cfg':
drivers/media/platform/xilinx/xilinx-dprxss.c:836:11: warning: 'i' is used uninitialized in this function [-Wuninitialized]
  836 |  format = v4l2_subdev_get_try_format(sd, cfg, i);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For below, initialize the i with 0.

drivers/media/platform/xilinx/xilinx-dprxss.c: In function 'xdprxss_probe':
drivers/media/platform/xilinx/xilinx-dprxss.c:1249:2: warning: 'i' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1249 |  for (j = 0; j < i; j++) {

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Rajesh Gugulothu <gugulothu.rajesh@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dprxss.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dprxss.c b/drivers/media/platform/xilinx/xilinx-dprxss.c
index 77aa00c20a6c..6c4127cebde9 100644
--- a/drivers/media/platform/xilinx/xilinx-dprxss.c
+++ b/drivers/media/platform/xilinx/xilinx-dprxss.c
@@ -836,9 +836,8 @@ static int xdprxss_init_cfg(struct v4l2_subdev *sd,
 {
 	struct xdprxss_state *xdprxss = to_xdprxssstate(sd);
 	struct v4l2_mbus_framefmt *format;
-	unsigned int i;
 
-	format = v4l2_subdev_get_try_format(sd, cfg, i);
+	format = v4l2_subdev_get_try_format(sd, cfg, 0);
 
 	if (!xdprxss->valid_stream)
 		*format = xdprxss->format;
@@ -1107,7 +1106,7 @@ static int xdprxss_probe(struct platform_device *pdev)
 	struct device *dev = &pdev->dev;
 	struct resource *res;
 	int ret, irq;
-	unsigned int i, j;
+	unsigned int i = 0, j;
 
 	xdprxss = devm_kzalloc(dev, sizeof(*xdprxss), GFP_KERNEL);
 	if (!xdprxss)
-- 
2.31.1

